include/zephyr/settings/settings.h

```diff
@@ -196,22 +202,30 @@ struct settings_handler_static {
  * @param _set set routine (can be NULL)
  * @param _commit commit routine (can be NULL)
  * @param _export export routine (can be NULL)
+ * @param _cprio commit priority (lower value is higher priority)
  *
  * This creates a variable _hname prepended by settings_handler_.
  *
  */
 
-#define SETTINGS_STATIC_HANDLER_DEFINE(_hname, _tree, _get, _set, _commit,   \
-				       _export)				     \
+#define SETTINGS_STATIC_HANDLER_DEFINE_WITH_CPRIO(_hname, _tree, _get, _set, \
+						  _commit, _export, _cprio)  \
 	const STRUCT_SECTION_ITERABLE(settings_handler_static,		     \
 				      settings_handler_ ## _hname) = {       \
 		.name = _tree,						     \
+		.cprio = _cprio,					     \
 		.h_get = _get,						     \
 		.h_set = _set,						     \
 		.h_commit = _commit,					     \
 		.h_export = _export,					     \
 	}
 
+/* Handlers without commit priority are set to priority O */
+#define SETTINGS_STATIC_HANDLER_DEFINE(_hname, _tree, _get, _set, _commit,   \
+				       _export)				     \
+	SETTINGS_STATIC_HANDLER_DEFINE_WITH_CPRIO(_hname, _tree, _get, _set, \
+		_commit, _export, 0)
```

Comment:
When I looked at `settings_commit_subtree`. The starting `int cprio = INT_MIN`; but the default is `0`, which will cause all the code that uses this PR to run slower once this PR is merged (although it seems trivial), but I feel it is best to change the default `0` to `INT_MIN`, because this will not affect the ready-made code.